platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/excludes/simple/DefaultExcludeFactory.java

```diff
@@ -60,22 +58,22 @@ public ModuleIdExclude moduleId(ModuleIdentifier id) {
 
     @Override
     public ExcludeSpec anyOf(ExcludeSpec one, ExcludeSpec two) {
-        return DefaultExcludeAnyOf.of(ImmutableSet.of(one, two));
+        return DefaultExcludeAnyOf.of(PersistentSet.of(one, two));
     }
 
     @Override
     public ExcludeSpec allOf(ExcludeSpec one, ExcludeSpec two) {
-        return DefaultExcludeAllOf.of(ImmutableSet.of(one, two));
+        return DefaultExcludeAllOf.of(PersistentSet.of(one, two));
     }
 
     @Override
-    public ExcludeSpec anyOf(Set<ExcludeSpec> specs) {
-        return DefaultExcludeAnyOf.of(ImmutableSet.copyOf(specs));
+    public ExcludeSpec anyOf(PersistentSet<ExcludeSpec> specs) {
+        return DefaultExcludeAnyOf.of(PersistentSet.copyOf(specs));
```

Comment:
copyOf should not be necessary here and below

Reply:
Right!